### PR TITLE
Start a new "email" ConnectionConfig type [#1134]

### LINKED ISSUE #1134: Create Fidesops Email Connector and Example Email Dataset
### Is your feature request related to a specific problem?

Add the scaffolding for a new EmailConnector that will eventually send an email to a user to complete an action, instead of accessing a database or a third party endpoint automatically.

### Describe the solution you'd like

- Create a Fidesops `EmailConnector` class (inheriting from `BaseConnector`
  - Stub out query_config, create_client, close, test_connection, retrieve_data, and mask_data for now.
- Add a schema migration that adds an "email" ConnectionType, so we can create `email` ConnectionConfigs
- Flesh out an EmailSchema describing the types of secrets we should add to the Email ConnectionConfig.  This will probably be very similar or identical to the `EmailConfigRequest` schema.
- Create an example "email" dataset for use in testing.  This will be similar to the existing "manual" dataset.
- Add API tests verifying that we can create `email` ConnectionConfigs and add secrets to to the ConnectionConfig 

### Describe alternatives you've considered, if any

_A description of any alternative solutions or features you've considered._

### Additional context

_Add any other context or screenshots about the feature request here._

### PR DESCRIPTION
**Pull request by pattisdr**
👉  Currently undocumented, the EmailConnector isn't fully fleshed out yet, that will come in a follow-up PR.
❗ Contains migration

# Purpose

Add the foundation for a new `email` ConnectionConfig type.  EmailConnector doesn't work yet here, that will be added in a follow-up PR.

# Changes
- Add a schema migration to add `email` to the ConnectionType enum
-  The expected secrets are is a `to_email` for the user that must be notified and an optional `test_email` to be used for the connection test 
- Add the beginnings of an `EmailConnector` class with most methods stubbed out.
- Add an example dataset for future testing 

# Checklist
- [X] Update [`CHANGELOG.md`](https://github.com/ethyca/fidesops/blob/main/CHANGELOG.md) file
  - [ ] Merge in main so the most recent `CHANGELOG.md` file is being appended to
  - [ ] Add description within the `Unreleased` section in an appropriate category. Add a new category from the list at the top of the file if the needed one isn't already there.
  - [ ] Add a link to this PR at the end of the description with the PR number as the text. example: [#1](https://github.com/ethyca/fidesops/pull/1)
- [ ] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
- [x] Good unit test/integration test coverage
- [x] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #1134

### CODE REVIEW

File: data/dataset/email_dataset.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            data_categories: [user.childrens]
+            fidesops_meta:
+              data_type: string
+          - name: parent_id
+            data_categories: [user]
+            fidesops_meta:
+              references:
+                - dataset: email_dataset
+                  field: daycare_customer.id
+                  direction: from
```
Comment (pattisdr) [inline data/dataset/email_dataset.yml:39]: Contrived dataset that will be used for testing in a followup PR.

File: src/fidesops/ops/schemas/connection_configuration/connection_secrets_email.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+)
+
+
+class EmailSchema(ConnectionConfigSecretsSchema):
+    """Schema to validate the secrets needed for the EmailConnector"""
+
+    to_email: str
+    test_email: Optional[str]  # Email to send a connection test email
+
+    _required_components: List[str] = ["to_email"]
```
Comment (pattisdr) [inline src/fidesops/ops/schemas/connection_configuration/connection_secrets_email.py:15]: currently I'm thinking the user can just configure the single email that should be sent out (or we could do a list of emails, that's in an earlier commit).  Additionally, if they have a "test_email" configured, I think they could send an email to themselves if they wanted with the same template they'd send to the user, just without the data in it, so they could preview if they wished.

File: src/fidesops/ops/service/connectors/email_connector.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        return []
+
+    def mask_data(  # type: ignore
+        self,
+        node: TraversalNode,
+        policy: Policy,
+        privacy_request: PrivacyRequest,
+        rows: List[Row],
+    ) -> Optional[int]:
+        """Stub"""
```
Comment (pattisdr) [inline src/fidesops/ops/service/connectors/email_connector.py:50]: This is all being fleshed out in a follow-up PR.  "retrieve_data" won't do anything, "mask_data" will figure out what data we need to send to the user.

Normally "mask_data" has rows retrieved from the access portion, but we won't have that for an email connector, so the email we send will need to be a combination of access/erasure information, to both tell them how to locate the data (access) and how/which fields to mask.

File: CHANGELOG.md [hunk 4]
```diff
@@ -19,6 +19,10 @@ The types of changes are:
 
 ## [Unreleased](https://github.com/ethyca/fidesops/compare/1.7.1...main)
 
+### Added
+
+* Foundations for a new email connector type [#1142] https://github.com/ethyca/fidesops/pull/1142
```
Comment (seanpreston) [inline CHANGELOG.md:24]: The URL is missing brackets here

File: src/fidesops/ops/schemas/connection_configuration/connection_secrets_email.py (hunk 2 above)
Comment (pattisdr) [reply src/fidesops/ops/schemas/connection_configuration/connection_secrets_email.py:15]: Current design just has this separately creating the single `EmailConfig` that Catherine developed, and using that resource to do the sending part.

File: CHANGELOG.md (hunk 4 above)
Comment (pattisdr) [reply CHANGELOG.md]: whoops

File: src/fidesops/ops/schemas/connection_configuration/connection_secrets_email.py (hunk 2 above)
Comment (seanpreston) [reply src/fidesops/ops/schemas/connection_configuration/connection_secrets_email.py:15]: To one `email` address is a fair assumption for now! And good thoughts on the test email for testing the connector.

**Review by seanpreston — APPROVED** [review]

**Pull request merged**